src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2766 @@
+    | -- | The node is configured with baker credentials and consensus is running.
+      NodeActive !BakerConsensusInfo
+
+instance FromProto Proto.NodeInfo'NetworkInfo where
+    type Output' Proto.NodeInfo'NetworkInfo = NetworkInfo
+    fromProto nInfo = do
+        nodeId <- nInfo ^? ProtoFields.nodeId . ProtoFields.value
+        peerTotalSent <- fromIntegral <$> nInfo ^? ProtoFields.peerTotalSent
+        peerTotalReceived <- fromIntegral <$> nInfo ^? ProtoFields.peerTotalReceived
+        avgBpsIn <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsIn
+        avgBpsOut <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsOut
+        return NetworkInfo{..}
+
+-- |Network related information of the node.
+data NetworkInfo = NetworkInfo
+    { -- |The node id.
+      nodeId :: !Text,
+      -- |Total number of packets sent by the node.
+      peerTotalSent :: !Word64,
+      -- |Total number of packets received by the node.
+      peerTotalReceived :: !Word64,
+      -- |Average outbound throughput in bytes per second.
+      avgBpsIn :: !Word64,
+      -- |Average inbound throughput in bytes per second.
+      avgBpsOut :: !Word64
+    }
+
+-- |Various information about the node.
+data NodeInfo = NodeInfo
+    { -- |The version of the node.
+      peerVersion :: !Text,
+      -- |The local time of the node.
+      localTime :: !Timestamp,
+      -- | Number of milliseconds that the node has been alive.
+      peerUptime :: !Duration,
+      -- | Information related to the p2p protocol.
+      networkInfo :: !NetworkInfo,
+      -- | Various details about the node.
+      details :: !NodeDetails
+    }
+
+instance FromProto Proto.ChainParametersV0 where
+    type Output' Proto.ChainParametersV0 = ChainParameterOutput
+    fromProto cParams = do
+        _cpElectionDifficulty <- fromProto =<< cParams ^? ProtoFields.electionDifficulty
+        _cpExchangeRates <- do
+            euroPerEnergy <- fromProto =<< cParams ^? ProtoFields.euroPerEnergy
+            microCcdPerEuro <- fromProto =<< cParams ^? ProtoFields.microCcdPerEuro
+            return $ Parameters.makeExchangeRates euroPerEnergy microCcdPerEuro
+        _cpCooldownParameters <- do
+            epoch <- fromProto =<< cParams ^? ProtoFields.bakerCooldownEpochs
+            return $ Parameters.CooldownParametersV0 epoch
+        let _cpTimeParameters = Parameters.TimeParametersV0
+        _cpAccountCreationLimit <- fromProto =<< cParams ^? ProtoFields.accountCreationLimit
+        _cpRewardParameters <- do
+            _rpMintDistribution <- fromProto =<< cParams ^? ProtoFields.mintDistribution
+            _rpTransactionFeeDistribution <- fromProto =<< cParams ^? ProtoFields.transactionFeeDistribution
+            _rpGASRewards <- fromProto =<< cParams ^? ProtoFields.gasRewards
+            return Parameters.RewardParameters{..}
+        -- FIXME: ProtoFields.foundationAccount has type AccountAddress, but the type expects an AccountIndex.
+        --        Is it possible to convert this?
```

Comment:
Not without context. You can look up an address via an index using `GetAccountInfo`. But you cannot do this without a network call since it depends on node state.

You should probably return an account index here, and a closure that takes an account address and returns chain parameters.

Then you can query account info together with chain parameters to get all the relevant data.

Reply:
I see. Since this is a non-optional field, should it be set to a dummy value then?

Reply:
Sounds reasonable.